### PR TITLE
Split transaction info out of env into a logical unit

### PR DESCRIPTION
**Pull request by pattyshack**

### CODE REVIEW

**Review by SaveTheRbtz — APPROVED** [review]

File: fvm/environment/transaction_info.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import (
+	"github.com/onflow/cadence/runtime"
+
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/trace"
+)
+
+type TransactionInfo interface {
```
Comment (zhangchiqing) [inline fvm/environment/transaction_info.go:11]: could you add some comments for the purpose/responsibility of this abstracted module

File: fvm/environment/transaction_info.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	SigningAccounts() []runtime.Address
+
+	IsServiceAccountAuthorizer() bool
+
+	// Candence's runtime api.
+	GetSigningAccounts() ([]runtime.Address, error)
```
Comment (zhangchiqing) [inline fvm/environment/transaction_info.go:17]: better to mention the possible error return would be `OperationNotSupportedError`

**Review by zhangchiqing — APPROVED** [review]
Looks good. Jus have some minor suggestion to the comments

File: fvm/environment/transaction_info.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+func (info *transactionInfo) GetSigningAccounts() ([]runtime.Address, error) {
+	defer info.tracer.StartExtensiveTracingSpanFromRoot(trace.FVMEnvGetSigningAccounts).End()
+
+	return info.authorizers, nil
+}
+
+var _ TransactionInfo = NoTransactionInfo{}
+
+type NoTransactionInfo struct {
```
Comment (janezpodhostnik) [inline fvm/environment/transaction_info.go:66]: A comment here that this is used in scripts would be nice as well.

**Review by janezpodhostnik — APPROVED** [review]
Looks good, but some comments about the purpose of interfaces/structs and uses would be nice.

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors retry

**Pull request merged**